src/arch/src/x86_64/interrupts.rs

```diff
@@ -27,6 +27,7 @@ const APIC_LVT1: usize = 0x360;
 const APIC_MODE_NMI: u32 = 0x4;
 const APIC_MODE_EXTINT: u32 = 0x7;
 
+#[allow(clippy::cast_sign_loss)] // bindgen generates [i8] but api wants u32
```

Comment:
In this case you can do `u32::try_from(byte_order::read_le_i32(reg)).unwrap()` instead of `byte_order::read_le_i32(reg) as u32`

Reply:
I tried that, the problem is that the registers are of type c_char (aka i8)

Reply:
Ah okay, you can do `u32::try_from(byte_order::read_le_i32(reg)).unwrap()` in this case.

Reply:
Wouldn't that be bad? In a scenario where someone writes an i32::max +1 into the register?
Here this test case would fail!
```rust
    #[test]
    fn test_set_and_get_klapic_reg_overflow() {
        let reg_offset = 0x340;
        let mut klapic = kvm_lapic_state::default();
        set_klapic_reg(&mut klapic, reg_offset, u32::try_from(i32::MAX).unwrap() + 1u32);
        let value = get_klapic_reg(&klapic, reg_offset);
        assert_eq!(value, u32::try_from(i32::MAX).unwrap() + 1u32);
    }
```

Reply:
The previous implementation with `as` appears to also result in an error in this test case. The easiest solution here is to add a `byte_order` functions that converts `&[i8]` to `u32`.

Reply:
Do you wish me to do that within this PR, or should I create a new issue?

Reply:
I think this fits in this PR.

Reply:
Could you double-check that you are not using the try_from version (within `get_klapic_reg`)? The test case does not fail for me, and more importantly, I think using the cast is correct here! 
I agree that changing the type of the registers to u8 would remove the cast, but the i8 type comes from the KVM binding.